extensions/sql-migration/src/api/utils.ts

```diff
@@ -261,3 +256,545 @@ export function clearDialogMessage(dialog: window.Dialog): void {
 export function getUserHome(): string | undefined {
 	return process.env.HOME || process.env.USERPROFILE;
 }
+
+export async function getAzureAccounts(): Promise<azdata.Account[]> {
+	let accounts: azdata.Account[] = [];
+	try {
+		accounts = await azdata.accounts.getAllAccounts();
+	} catch (e) {
+		console.log(e);
+		accounts = [];
+	}
+	return accounts;
+}
+
+export async function getAzureAccountsDropdownValues(accounts: azdata.Account[]): Promise<azdata.CategoryValue[]> {
+	let accountsValues: azdata.CategoryValue[] = [];
+	accounts.forEach((account) => {
+		accountsValues.push({
+			name: account.displayInfo.userId,
+			displayName: account.isStale
+				? constants.ACCOUNT_CREDENTIALS_REFRESH(account.displayInfo.displayName)
+				: account.displayInfo.displayName
+		});
+	});
+	if (accountsValues.length === 0) {
+		accountsValues = [
+			{
+				displayName: constants.ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR,	//
+				name: ''
+			}
+		];
+	}
+	return accountsValues;
+}
+
+export async function getAzureTenants(account?: azdata.Account): Promise<Tenant[]> {
+	let tenants: Tenant[] = [];
+	try {
+		if (account) {
+			tenants = account.properties.tenants;
+		}
```

Comment:
this seems excessive, to wrap account?.properties?.tenants with 12 lines of code.  It should not be needed or async

Reply:
Changed to be non async, agreed it's a little excessive but I think it's best to leave it in so that we have `getAccounts`, `getTenants`, `getManagedInstances`, etc. all in one place whenever we need to use it